scripts/fb_reader.py

```diff
@@ -0,0 +1,158 @@
+import sys
+import sqlite3
+import os
+import json
+import datetime
+import urllib
+
+from scripts.utils import OUTPUT_INIT
+
+''' location of databases '''
+threadsdb = OUTPUT_INIT + "/db/threads_db2"
+contactsdb = OUTPUT_INIT + "/db/contacts_db2"
+
+
+def store_fb_contacts():
+    if os.path.isfile(contactsdb):
+        contacts_conn = sqlite3.connect(contactsdb)
+    else:
+        print("contactsdb does not exist")
+        return
+
+    contacts_cursor = contacts_conn.cursor()
+    contacts_query = "select contact_id, data from contacts ;"
+    contacts_cursor.execute(contacts_query)
+
+    CONTACTS_CONTACTS_ID_INDX = 0
+    CONTACTS_DATA_INDX = 1
+    contacts_dict = {}
+
+    row = contacts_cursor.fetchone()
+    while row:
+        try:
+            extracted_data = json.loads(row[CONTACTS_DATA_INDX])
+        except:
+            print("Could not get data for contact_id = " + row[CONTACTS_CONTACTS_ID_INDX])
+            row = contacts_cursor.fetchone()
+            continue
+
+        dis_num = "None"
+        univ_num = "None"
+        if len(extracted_data["phones"]):
+            dis_num = extracted_data["phones"][0]["displayNumber"]
+            univ_num = extracted_data["phones"][0]["universalNumber"]
+
+        contacts_dict[row[0]] = (
+        extracted_data["profileFbid"], extracted_data["name"]["displayName"], dis_num, univ_num,
+        extracted_data["hugePictureUrl"])
+        row = contacts_cursor.fetchone()
+
+    contacts_cursor.close()
+    contacts_conn.close()
+
+    CONTACTS_PROFILE_ID_INDX = 0
+    CONTACTS_DISPLAY_NAME_ID_INDX = 1
+    CONTACTS_DISPLAY_PHONE_INDX = 2
+    CONTACTS_UNIV_PHONE_INDX = 3
+    CONTACTS_HUGE_PIC_URL_INDX = 4
+
+    sorted_contact_dict = sorted(contacts_dict, key=lambda x: contacts_dict[x][CONTACTS_DISPLAY_NAME_ID_INDX])
+
+    contacts_output_file = "contacts.tsv"
+
+    contacts_file = open(contacts_output_file, "w+", encoding="utf-8")
+    contacts_file.write("contact_id\tprofileFbid\tdisplayName\tdisplayNumber\tuniversalNumber\thugePictureUrl\n")
```

Comment:
Are we storing specific data in tsv's or few details only?

Reply:
```suggestion
    contacts_file.write("contact_id\tprofileFbid\tdisplayName\tdisplayNumber\tuniversalNumber\thugePictureUrl\n")
```
this is storing few important information from `.db` file. Mainly data queries will be on SQLite's